dffml/operation/archive.py

```diff
@@ -0,0 +1,74 @@
+import uuid
+from zipfile import ZipFile
+from pathlib import Path
+from typing import Union, Dict, Any, Optional
+
+from ..df.base import op
+from ..df.types import Definition
+
+
+# definitions
+PATH = Definition(name="path", primitive="Union[Path, str]")
+OPTIONS = Definition(name="options", primitive="Dict[str, Any]")
+
+
+@op(
+    inputs={
+        "input_directory_path": PATH,
+        "output_file_path": PATH,
+        "options": OPTIONS,
+    },
+    outputs={},
+)
+async def make_zip_archive(
+    input_directory_path: Union[Path, str],
+    output_file_path: Optional[Union[Path, str]] = None,
+    **options: Optional[Dict[str, Any]],
+):
+    """
+    Creates zip file of a directory.
+
+    Parameters
+    ----------
+    input_directory_path : Union[Path, str]
+        Path to directory to be archived
+    output_file_path : Optional[Union[Path, str]], optional
+        If set output file would be saved to this path otherwise 
+        file is saved in the given directory, by default None
+    """
+    if output_file_path is None:
+        file_name = str(uuid.uuid4()) + ".zip"
+        output_file_path = Path(input_directory_path) / file_name
+    with ZipFile(output_file_path, "w", **options) as zip:
+        for file in input_directory_path.rglob("*"):
+            zip.write(file)
```

Comment:
For testing, let's use `unittest.mock` rather than creating archives.

Reasoning behind this is that some archive formats can be trickey to make reproducible archives (for example, see `TestCachedDownloadServer`)